### PR TITLE
LLT-3951 IPv6 analytics

### PR DESCRIPTION
**Pull request by matislovas**
### Problem
We need to collect analytics on IPv6 usage. But current WG implementations does not distinguish data between IP protocols.

### Solution
The following solution is being used:

- Add IPv4/IPv6 flags/bits in connectivity matrix.
- Add `rtt6` collection, using `ICMPv6` protocol. Additionally, to get a more clear picture of IP protocol health, `rtt_loss` / `rtt6_loss`  which collects percentages of lost `ICMP` / `ICMPv6` packets.

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] changelog.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

**Comment by matislovas** [comment]
I see that there's no nat-lab tests for `QoS` collection. I mean, I can try to add some, but maybe there's a reason, why the were omitted in the first place ..?

**Comment by matislovas** [comment]
This should be merged in sync with `libtelio-build` !134 .

**Comment by matislovas** [comment]
The nat-lab tests still fails, waiting for moose release with added additional metrics.

File: nat-lab/tests/test_lana.py
Comment (arvydasv) [inline nat-lab/tests/test_lana.py:1149]: Create a variable for this to remove repeating code.

**Review by arvydasv — COMMENTED** [review]
+1

File: nat-lab/tests/test_lana.py
Comment (matislovas) [reply nat-lab/tests/test_lana.py]: Done!

File: .github/workflows/gitlab.yml
Comment (Jauler) [inline .github/workflows/gitlab.yml:34]: Adding a message not to forget to change this to actual tag once that is merged in `libtelio-build`

File: .gitlab-ci.yml
Comment (Jauler) [inline .gitlab-ci.yml:18]: Same as https://github.com/NordSecurity/libtelio/pull/144/files#r1360759322

File: crates/telio-nurse/src/qos.rs
Comment (Jauler) [inline crates/telio-nurse/src/qos.rs:463]: Not sure what was agreed, but should'nt `loss` be represented as just simple percentage number? Or event attempts vs success counts (this way we will know if we get 50% because we lost one packet, but attempted only twice, or whether we genuinely have 50% loss.

Comment (matislovas) [reply crates/telio-nurse/src/qos.rs:463]: Yup, but my impression here was, that these buckets are formed from individual ping sessions (it represents the loss percentage `20%:40%:70%:90%:100%`).

File: crates/telio-nurse/src/heartbeat.rs
Comment (Jauler) [reply crates/telio-nurse/src/heartbeat.rs:439]: We have discussed live, and I realised that this represents desired stack and then rtt and rtt_loss indicates whether we actually have anything there. Therefore I will mark this as resolved as it is doing as it is supposed to do.

File: changelog.md
Comment (baltrus1) [inline changelog.md:34]: deleted while rebase?

Comment (matislovas) [reply changelog.md:34]: Whoops, fixed!

File: crates/telio-nurse/src/rtt/ping.rs
Comment (baltrus1) [inline crates/telio-nurse/src/rtt/ping.rs:124]: isn't .0 is always ipv4 and .1 is always ipv6?

**Review by baltrus1 — APPROVED** [review]
+1

File: crates/telio-nurse/src/rtt/ping.rs
Comment (matislovas) [reply crates/telio-nurse/src/rtt/ping.rs:124]: Nope, on IPv6 stack, the primary address is IPv6:

```
if let (Some(ip4), Some(ip6)) = (self.target.0, self.target.1) {
    // IPv6 target is preffered, because we can be sure, that address is unique
    Ok((IpAddr::V6(ip6), Some(IpAddr::V4(ip4))))
} else if let (Some(ip4), None) = (self.target.0, self.target.1) {
    Ok((IpAddr::V4(ip4), None))
} else if let (None, Some(ip6)) = (self.target.0, self.target.1) {
    Ok((IpAddr::V6(ip6), None))
} else {
    Err(DualTargetError::NoTarget)
}
```

**Review by packgron — APPROVED** [review]

**Pull request merged**
